src/uct/ib/base/ib_device.c

```diff
@@ -698,7 +699,27 @@ int uct_ib_device_test_roce_gid_index(uct_ib_device_t *dev, uint8_t port_num,
     return 1;
 }
 
-ucs_status_t uct_ib_device_select_gid(uct_ib_device_t *dev, uint8_t port_num,
+int uct_ib_device_is_potentially_reachable(unsigned subnet_mask,
+                                           const uct_ib_device_gid_info_t *gid_info)
+{
+    size_t addr_size;
+
+    if (gid_info->roce_info.ver == UCT_IB_DEVICE_ROCE_V1) {
+        return 1;
+    }
+
+    /*
+     * If the subnet mask covers the entire address, that means we're in a
+     * non-routable subnet and should only use RoCEv1.
```

Comment:
why? we still may want to use RoCEv2 for things like DCQCN

Reply:
If there's a reachability issue (like I have in my scenario) - how can we still use RoCEv2 for anything? packets will not reach the destination :)

Reply:
why would using RoCEv1 instead of RoCEv2 fix a reachability issue?

Reply:
I'm still investigating this, maybe it has to do with socket-direct in our setup, but the fact is that out-of-the-box RoCEv2 gives out some timeout error during connection establishment, whereas RoCEv1 works. It could also be switch configuration, but unlikely.

Reply:
@yosefe I got back to this issue today, and I think I found the root cause: #5598 
I can implement the fix, but I suppose it'll be simpler if I do it in a separate PR - once this one is done.